docs/release/release_0_4_16.md

```diff
@@ -26,15 +26,37 @@ After discussion in [#4102](https://github.com/napari/napari/pull/4102) and [#41
 
 
 - Calling `viewer.open` *without* passing a plugin will result in an error if you have not saved a reader preference for that file pattern *and* multiple plugins can claim the file
-    - You can address this error by associating a preference for the file pattern, or calling `viewer.open(file_path, plugin=...)
-- A preferred reader failing to read your file will result in an error
+    - To save a preference for a file pattern at the command line:
+
+    ```python
+    from napari.settings import get_settings
+    get_settings().plugins.extension2reader['*.tif'] = napari_tifffile
+    get_settings().plugins.extension2reader['*.zarr'] = napari_ome_zarr
+    ```
+    - To pass a plugin in a Python script:
+
+    ```python
+    import napari
+
+    viewer = napari.Viewer()
+    viewer.open('my-path.tif') # this will throw MultipleReaderError if napari_tifffile is installed as both it and builtins could open the file
+    viewer.open('my-path.tif', plugin=napari_tifffile) # this won't
+    ```
+
+    - To pass a plugin at the command line
+    
+    ```sh
+    napari my-path.tif --plugin napari_tifffile
+    ```
+- A preferred reader failing to read your file will result in an error e.g. if you saved `napari_tifffile` as a preference for TIFFs but then uninstalled it
```

Comment:
```suggestion
    ```

- A preferred reader failing to read your file will result in an error e.g. if you saved `napari_tifffile` as a preference for TIFFs but then uninstalled it
```

I didn't realise this was the case @DragaDoncila and I think it's a big problem, because napari doesn't have per-environment preferences. imho plugin preferences that don't match an installed plugin should be ignored...

Reply:
Two things I guess, @jni.

- If a preferred plugin is available but just fails to read the file for some reason we raise `ReaderPluginError`, I think you can agree that's not controversial
- If a preferred plugin is missing, we raise `MissingAssociatedReaderError` which is the bit I think you'd like to change. Idk how I feel about that. I actually didn't realize settings weren't per environment (though I should've), but I don't know if ignoring preferences is the right way? I feel like it could be confusing to associate a reader, it's not found, and we just ignore that entirely. Especially since you can't add two preferences for the same file type. Maybe we should warn? Idk I'd love more eyes on this @tlambert03 @nclack? I think if we want to change that pre-release it should be a separate PR anyway.